reference/language-concepts/augments.markdown

```diff
@@ -6,45 +6,172 @@ sorting: 70
 tags: [manuals, language, syntax, concepts, augments]
 ---
 
-An augments file can be used to define variables and classes to the execution of
-**all** CFEngine components **before** any parsing or evaluation happen. It's a
-JSON data file, so you should view and edit it with a JSON-aware editor if
-possible.This is a convenient way to override defaults defined in the
+Augments files can be used to define variables and classes for use by
+**all** CFEngine components **before** any parsing or evaluation happen. Augments are fundamentally
+JSON data files, so you should view and edit them with a JSON-aware editor if
+possible. This is a convenient way to override defaults defined in the
 Masterfiles Policy Framework without modifying the shipped policy itself.
 
-The file `def.json` is found like the policy file to be run:
+There are two canonical augments files, `$(sys.workdir)/data/host_specific.json` (typically `/var/cfengine/data/host_specific.json`), which is read first and who's Variables take precedence, and `$(sys.policy_entry_dirname)/def.json` (typically `/var/cfengine/inputs/def.json`) which may load additional Augments as specified by the `augments` key.
+
+The file `def.json` is found based on the location of the policy entry (the first policy file read by the agent):
 
 * with no arguments, it's in `$(sys.inputdir)/def.json` because
   `$(sys.inputdir)/promises.cf` is used
 * with `-f /dirname/myfile.cf`, it's in `/dirname/def.json`
 * with `-f ./myfile.cf`, it's in `./def.json`
 
+`def_preferred.json` will be used instead of `def.json` if it is present. This preferential loading can be disabled by providing the `--ignore-preferred-augments` option to the agent.
+
 Values will be expanded, so you can use the variables from
 [Special Variables][].
 
 An augments file can contain the following keys:
 
 ## inputs
 
+This key is supported in `def.json`, `def_preferred.json`, and augments loaded by the augments key.
+
 Any filenames you put here will appear in the `def.augments_inputs` variable.
 The standard set of masterfiles refers to this variable and will autoload those
 files.
 
-## vars
+## variables
+
+This key is supported in both `host_sepecific.json`, `def.json`, `def_preferred.json`, and augments loaded by the augments key.
+
+Variables defined here can target a _namespace_ and or _bundle_ scope explicitly. When defined from `host_specific.json`, variables default to the `main` _bundle_ in the `data` __namespace_ (`$(data:main.MyVariable)`).
+
+For example:
+
+```json
+{
+    "variables": {
+        "VariableWithImplicitNamespaceAndBundle": "value",
+        "comment": "An optional note about why this variable is important",
+        "tags": [ "optional", "tags" ]
```

Comment:
@nickanderson @craigcomstock @vpodzime This is not how it works, right? AFAIK `comment` and `tags` need to be inside an object.